view/propertyEditor.js

```diff
@@ -245,77 +244,143 @@ ACS.propertyEditor = function (modelList, // ACS.modelList
 				cell.appendChild(dropdownList);
 			}
 
-			//generate checkbox field for boolean
-			if (comboValues === '' && typetemp === ACS.dataType.BOOLEAN) {
+			if (isDynProp) {
+
+				log.debug('downloading (dynamic) properties list');
 				cell = row[h + 1].insertCell(1);
-				boolInput = null;
-				boolInput = document.createElement("INPUT");
-				boolInput.setAttribute("type", "checkbox");
-				boolInput.setAttribute("value", valtemp);
-				if ((valtemp === "true") || (valtemp === "True")) {
-					boolInput.setAttribute("checked", true);
+
+				if (ACS.areStatus.getSynchronised() === true) {
+
+					var id = propertyEdPanelCaptions.innerHTML;
+					var key = propName;
+					var currentValue = valtemp || "";
+					var elementId = h + "/1/" + valtemp;
+	
+					fetchDynProperties(cell, elementId, currentValue, id, key);
+					function fetchDynProperties(cellToAdd, elementId, currentValue, id, key) {
+						getRuntimeComponentPropertyList(function(data, httpStatus) {
+							var entries = data ? JSON.parse(data) : null;
+							if (entries === null || !(entries instanceof Array) || entries.length === 0) {
+								if (entries === null) console.warn('cannot process received data');
```

Comment:
so, new day - new try to review :D
(1) `(entries === null || !(entries instanceof Array) || entries.length === 0)` could be simplified to
 `(!(entries instanceof Array) || entries.length === 0)` since `(null instanceof Array) == false`

(2) `if (entries === null) console.warn('cannot process received data');`
is semantically not correct, because if the received data could not be processed `JSON.parse(data)` above would throw an error and this line would not be reached. If you want to catch JSON parsing errors you would have to  surround `JSON.parse()` with try/catch. However I don't know if this is necessary - depends how much we trust `ARE` to deliver correct data.

(3) I don't like if/else statements without curly braces.

If you are tired of changing these lines of code over and over again, for me it would be also OK to merge as-is, since functional it should be correct now.
Since

Reply:
I'm not tired of changing those lines :relaxed: Don't worry. I'll integrate (1) and (3). For (2) I'll change the text to "received null" (??). Or should I drop it entirely?

Reply:
new day, new insights!
I think we all focused onto the wrong line!!

actually ```data``` can be null and this leads to a filled entries array or null.
So why not simply checking data in the if condition and to the rest within the if-body?

```javascript
getRuntimeComponentPropertyList(function(data, httpStatus) {
    if(data === null) {
        // insert empty field
    } else {
       entries=JSON.parse(data);
       //insert dropbox
   }
```

Reply:
@deinhofer: the problem with your solution is that `JSON.parse(data)` can also be an empty array and in this case we also want to insert the empty field.

@sabicalija: for (2) I would just simply do `console.warn('no dynamic properties from ARE - using textfield instead of combobox')` and not distinguish between the two cases.

Maybe the cleanest solution would be a separate function:
```
function parseEntries (jsonString) {
    try{
        let parsed = JSON.parse(jsonString);
        return parsed instanceof Array ? parsed : [];
    } catch(error) {}
    return [];
}
```

and then just:
```
getRuntimeComponentPropertyList(function(data, httpStatus) {
    let entries = parseEntries(data);
    if(entries.length === 0) {
        // insert empty field
    } else {
       //add empty element to entries
       //insert dropbox
   }
```

Reply:
the function `parseEntries` could also be generalized to something like `parseJSONtoArray()` and then added to some util class, if any is available.

Reply:
Proposed changes are now added (see https://github.com/asterics/WebACS/pull/36/commits/52a208603f7c636d134dfe23eebc81371ce12633). Should be fine for now. However, I'd like to use general `parseJSONtoArray()` in future.